prototyping/pyegsnrc/particles.py

```diff
@@ -0,0 +1,15 @@
+from typing import Dict
+from typing_extensions import Literal
+
+import jax.numpy as jnp
+
+ParticleKeys = Literal["position", "direction", "energy"]
+T = Dict[ParticleKeys, jnp.DeviceArray]
```

Comment:
What are your thoughts with treating the `particle` module like an object. Where the module's type is stored within the variable `T`.